Java/core/src/main/java/com/amazon/randomcutforest/state/RandomCutForestMapper.java

```diff
@@ -123,64 +123,65 @@
      */
     @Override
     public RandomCutForestState toState(RandomCutForest forest) {
-        if (saveTreeState) {
+        if (saveTreeStateEnabled) {
             checkArgument(forest.isCompact(), "tree state cannot be saved for noncompact forests");
         }
 
         RandomCutForestState state = new RandomCutForestState();
 
         state.setNumberOfTrees(forest.getNumberOfTrees());
         state.setDimensions(forest.getDimensions());
-        state.setLambda(forest.getLambda());
+        state.setTimeDecay(forest.getLambda());
         state.setSampleSize(forest.getSampleSize());
         state.setShingleSize(forest.getShingleSize());
         state.setCenterOfMassEnabled(forest.isCenterOfMassEnabled());
         state.setOutputAfter(forest.getOutputAfter());
         state.setStoreSequenceIndexesEnabled(forest.isStoreSequenceIndexesEnabled());
         state.setTotalUpdates(forest.getTotalUpdates());
-        state.setCompactEnabled(forest.isCompact());
+        state.setCompact(forest.isCompact());
         state.setInternalShinglingEnabled(forest.isInternalShinglingEnabled());
         state.setBoundingBoxCacheFraction(forest.getBoundingBoxCacheFraction());
-        state.setSaveSamplerState(saveSamplerState);
-        state.setSaveTreeState(saveTreeState);
-        state.setSaveCoordinatorState(saveCoordinatorState);
-        state.setSinglePrecisionSet(forest.getPrecision() == Precision.FLOAT_32);
-        state.setCompress(compress);
-        state.setPartialTreesInUse(partialTreesInUse);
-
-        if (saveExecutorContext) {
+        state.setSaveSamplerStateEnabled(saveSamplerStateEnabled);
+        state.setSaveTreeStateEnabled(saveTreeStateEnabled);
+        state.setSaveCoordinatorStateEnabled(saveCoordinatorStateEnabled);
+        state.setPrecision(forest.getPrecision().name());
+        state.setCompressed(compressionEnabled);
+        state.setPartialTreeState(partialTreeStateEnabled);
+
+        if (saveExecutorContextEnabled) {
             ExecutorContext executorContext = new ExecutorContext();
             executorContext.setParallelExecutionEnabled(forest.isParallelExecutionEnabled());
             executorContext.setThreadPoolSize(forest.getThreadPoolSize());
             state.setExecutorContext(executorContext);
         }
 
         if (forest.isCompact()) {
-            if (saveCoordinatorState) {
-                PointStoreCoordinator pointStoreCoordinator = (PointStoreCoordinator) forest.getUpdateCoordinator();
+            if (saveCoordinatorStateEnabled) {
+                PointStoreCoordinator<?> pointStoreCoordinator = (PointStoreCoordinator<?>) forest
+                        .getUpdateCoordinator();
                 PointStoreState pointStoreState;
                 if (forest.getPrecision() == Precision.FLOAT_32) {
```

Comment:
forest.getPrecision().name()?? It's probably ok in the model -> state direction.

Reply:
`precision` in the forest class has the enum type `Precision`, so comparing to the enum value `FLOAT_32` is correct here. The `.name()` returns the string representation of the enum constant. In the state classes, I made the `precision` field a String. My reasoning was that I didn't want to break protostuff if we added new enum values later (I don't know if that would break it or not, but it didn't seem worth taking the chance).

Reply:
makes sense.